### ISSUE TITLE
Fallback function in ABI interface JSON

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#499]
Shouldn't the fallback function be listed in there with a special type?

Suggestions:
- `name` = `(fallback)` or
- `fallback` field set to `true`

e.g. `{"constant":false, "inputs":[], "name":"(fallback)", "outputs":[], "type":"function"}`

Going forward (see https://github.com/ethereum/EIPs/issues/34) the plan is to either reject or allow the fallback function via modifiers in a contract. Having an explicit fallback method in the interface ABI would also signal that to the outside world.

Side notes:
- solc accepts the `constant` modifier on the fallback function
- it also accepts output parameters
- and rejects input parameters:

```
tests:2:14: Error: Fallback function cannot take parameters.
    function (uint a) constant returns (uint) {
             ^------^
```

If the constant modifier and output fields are allowed that is a bit more reason to have in the interface ABI. If they are an oversight and shouldn't be allowed, than it can be argued it doesn't belong to the interface.

**Comment by chriseth** [comment]
Return values in the fallback function is an oversight. Actually the mere existence of the fallback function is due to the fact that a contract can be called while ignoring the ABI, so I would argue that it should not appear in the ABI.

**Comment by axic** [comment]
Since the ABI is not part of the EVM, it puts the onus on the contract. Is there a plan to change that?

I think the fallback method is useful, but to get rid of them, you could:
- throw if the dispatcher doesn't find a matching method
- have a flag in the contract disabling the fallback (I think you have that on pivotaltracker somewhere)

I do like the fallback function. I am not fully sure it should or shouldn't have constant and output modifiers, but heavily leaning towards that it shouldn't.

Given it could be enabled or disabled via a keyword in the future, I really think it should have its own entry in the JSON ABI interface.

**Comment by chriseth** [comment]
The only reason for having fallback functions (instead of weird "manual override" cases) is to react (basically log an event) when receiving ether.

Since we plan to include the strict "payable" modifier, changes to the ABI could be limited to "can receive ether" for a contract, the signature of the fallback function does not convey any further information.

**Comment by axic** [comment]
Fixed in #664.
